src/WebJobs.Extensions.DurableTask/Listener/OutOfProcOrchestrationShim.cs

```diff
@@ -178,14 +179,26 @@ private async Task ProcessAsyncActions(AsyncAction[][] actions)
                             this.context.ContinueAsNew(action.Input);
                             break;
                         case AsyncActionType.WaitForExternalEvent:
-                            tasks.Add(this.context.WaitForExternalEvent<object>(action.ExternalEventName));
+                            newTask = this.context.WaitForExternalEvent<object>(action.ExternalEventName);
                             break;
                         case AsyncActionType.CallHttp:
-                            tasks.Add(this.context.CallHttpAsync(action.HttpRequest));
+                            newTask = this.context.CallHttpAsync(action.HttpRequest);
                             break;
                         default:
                             break;
                     }
+
+                    if (newTask != null)
+                    {
+                        // Check to see if the context API call failed e.g. because the caller tried to schedule a function that doesn't exist.
+                        if (newTask.IsFaulted)
+                        {
+                            // Awaiting a faulted task will cause the exception to be thrown.
```

Comment:
Just to make sure I understand the ramifications of this:

I would expect that this means that the Azure Functions execution would still have succeeded, but the orchestration will fail.

I think that is acceptable/the best we can do right now.

Reply:
Yes, the function execution would succeed but the orchestration would fail. It's not ideal but it's incrementally better than where we are today. I think we'll need to revisit aspects the out-of-proc design in order to make function executions fail. I have a couple ideas on strategies we could take but I'll save that for a different forum.

Reply:
Oops! I was wrong! It turns out that the function execution reports as failed even though the failure happens in the out-of-proc shim layer. I added logs in the PR description to illustrate. I suspect this is the case because the exception happens in the WebJobs middleware layer which I believe is part of the core function execution.